crates/optimism/primitives/src/transaction/signed.rs

```diff
@@ -71,6 +72,16 @@ impl OpTransactionSigned {
     pub const fn is_deposit(&self) -> bool {
         matches!(self.transaction, OpTypedTransaction::Deposit(_))
     }
+
+    /// Returns the estimated compressed size of a transaction.
+    pub fn da_usage(&self) -> u64 {
+        let mut tx_ser: Vec<u8> = Vec::new();
+        // TODO: check handling of deposit/blob txs - I think they're treated as 0 usage
+        self.transaction.legacy().unwrap().eip2718_encode(&self.signature, &mut tx_ser);
+
+        estimate_tx_compressed_size(&tx_ser)
```

Comment:
kind of unfortunate perf wise if we need to serialize every tx, right? should this be memoized somehow?

Reply:
agreed - is there a standard way to memoize serialized values? I added a simple memoized property to the struct here, but there might be a better way to do it. [d41de90](https://github.com/paradigmxyz/reth/pull/13757/commits/d41de90a7a2040d0d52484b6580fdb842e683681)

Reply:
we can do this if we change the `OpTransactionSigned` type and a lazy field for this there, however this wouldn't enough for the building use case because this operates on OpPooledTx mostly, though luckily we can already provide different types for this.

I need to think about this a bit, because I had some ideas how we can improve the builder interface.
and ideally also solve the issue with repeated 2718 encoding as well.